toolkit/recordreplay/ProcessRecordReplay.cpp

```diff
@@ -219,13 +220,41 @@ MOZ_EXPORT void RecordReplayInterface_Initialize(int* aArgc, char*** aArgv) {
   }
   MOZ_RELEASE_ASSERT(dispatchAddress.isSome());
 
+  Maybe<std::string> apiKey;
+  {
+#ifdef _WIN32
+    // Note: This code is not tested on Windows.
+    char buf[200];
+    DWORD rv = GetEnvironmentVariable("RECORD_REPLAY_API_KEY", buf, sizeof(buf));
+    if (rv != 0) {
+      if (rv > sizeof(buf)) {
+        char* data = new char[rv];
+        DWORD rv2 = GetEnvironmentVariable("RECORD_REPLAY_API_KEY", data, rv);
+        MOZ_RELEASE_ASSERT(rv2 + 1 === rv);
+        apiKey.emplace(data);
+        delete[] data;
+      } else {
+        apiKey.emplace(buf);
+      }
+      MOZ_RELEASE_ASSERT(SetEnvironmentVariable("RECORD_REPLAY_API_KEY", nullptr));
+    }
+#else
+    const char* val = getenv("RECORD_REPLAY_API_KEY");
+    if (val) {
+      apiKey.emplace(val);
+      MOZ_RELEASE_ASSERT(!unsetenv("RECORD_REPLAY_API_KEY"));
+    }
+#endif
+  }
+
   gDriverHandle = OpenDriverHandle();
   if (!gDriverHandle) {
     fprintf(stderr, "Loading driver failed, crashing.\n");
     MOZ_CRASH("RECORD_REPLAY_DRIVER loading failed");
   }
 
   LoadSymbol("RecordReplayAttach", gAttach);
+  LoadSymbol("RecordReplaySetApiKey", gSetApiKey, /* aOptional */ true);
```

Comment:
_Probably_ not useful to make this optional now that we bundle the driver, but wasn't 100% sure.

Reply:
Yeah, this doesn't need to be optional with the driver bundling.